### PR TITLE
[interop] AST printer should only visit one original namespace decl t…

### PR DESCRIPTION
**Pull request by hyp**
…o avoid printing redecls over and over again

Also, lookup namespace members directly without doing a lookup. This helps us print the libc++ module interface in < 1s on M1.

Also, print namespace redecls that are specific to the module being printed. This ensures that when we're printing multiple module interfaces, we separate out the namespaces and their members correctly between the produced module interfaces.

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by hyp** [comment]
@swift-ci please test macOS platform

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test

File: lib/AST/ASTPrinter.cpp [hunk 1]
```diff
@@ -14,8 +14,8 @@
 //
 //===----------------------------------------------------------------------===//
```
Comment (zoecarver) [inline lib/AST/ASTPrinter.cpp:15]: Unrelated to this patch: would be great to pull this tool into the frontend.

**Review by zoecarver — APPROVED** [review]
This is great! Thanks, Alex!

**Comment by hyp** [comment]
@swift-ci please test macOS platform

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by hyp** [comment]
@swift-ci please test macOS platform

**Comment by hyp** [comment]
@swift-ci please test macOS platform

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by hyp** [comment]
@swift-ci please test macOS platform

**Comment by hyp** [comment]
@swift-ci please test macOS platform

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by hyp** [comment]
source compatibility failure is unrelated:

fatal error: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't open input file: /Users/ec2-user/jenkins/workspace-private/swift-PR-source-compat-suite-debug-macos/build/compat_macos/swiftdriver-macosx-x86_64/x86_64-apple-macosx10.15/release/dependencies/swift-tools-support-core/lib/libTSCLibc.dylib (No such file or directory)

Going to merge.

File: test/Interop/Cxx/stdlib/print-libcxx-module-interface.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// RUN: %target-swift-ide-test -print-module -module-to-print=CxxStdlib -source-filename=x -enable-experimental-cxx-interop -enable-objc-interop -module-print-submodules | %FileCheck %s
```
Comment (egorzhdan) [inline test/Interop/Cxx/stdlib/print-libcxx-module-interface.swift:1]: Is this test different from `test/Interop/Cxx/stdlib/libcxx-module-interface.swift`?

**Pull request merged**